### PR TITLE
feat(node): token bridge set pauser gov

### PR DESCRIPTION
**Pull request by evan-gray**
Implemented the node portion of #4809 and only that, as opposed to #4769

- Design #4809
- Guardian #4810
- EVM WTT Pauser Role #4801
- SVM WTT Pauser Role #4802

### CODE REVIEW

File: node/pkg/adminrpc/adminserver.go [hunk 1]
```diff
@@ -310,6 +310,54 @@ func accountantModifyBalance(req *nodev1.AccountantModifyBalance, timestamp time
...
+		return nil, fmt.Errorf("failed to serialize governance body: %w", err)
+	}
+
+	return vaa.CreateGovernanceVAA(timestamp, nonce, sequence, guardianSetIndex, body), nil
+}
+
+// decodePauserAddress decodes a hex pauser/unpauser address string. An empty
+// string decodes to an empty slice, which the serializer encodes as a
+// zero-length address - i.e. the role being unassigned.
+func decodePauserAddress(field, s string) ([]byte, error) {
```
Comment (johnsaigle) [inline node/pkg/adminrpc/adminserver.go:350]: Wanna use sdk's StringToAddress instead? https://github.com/wormhole-foundation/wormhole/blob/4028572892483c79146c14df44993e46401a9b54/sdk/vaa/structs.go#L824-L848

The only difference is that it limits the length to 32 where the implementation here allows for 255. That seems better though IMO, and if we need addresses longer than 32 then there's a good case for updating the SDK to support that.

File: node/cmd/guardiand/admintemplate.go [hunk 2]
```diff
@@ -574,6 +593,81 @@ func runTokenBridgeUpgradeContractTemplate(cmd *cobra.Command, args []string) {
...
+
+	b, err := prototext.MarshalOptions{Multiline: true}.Marshal(m)
+	if err != nil {
+		log.Fatal("failed to marshal request: ", err)
+	}
+	fmt.Print(string(b))
+}
+
+// parsePauserAddressHex parses a pauser/unpauser address argument and returns
+// the lower-case hex encoding used on the wire (no 0x prefix). An empty input
```
Comment (johnsaigle) [inline node/cmd/guardiand/admintemplate.go]: I don't know if this necessarily returns the lower-case hex encoding .

```
go doc hex.DecodeString
package hex // import "encoding/hex"

func DecodeString(s string) ([]byte, error)
    DecodeString returns the bytes represented by the hexadecimal string s.

    DecodeString expects that src contains only hexadecimal characters and that
    src has even length. If the input is malformed, DecodeString returns the
    bytes decoded before the error.
```

I don't think it matters for this context though so this comment might be a misleading by how precise it is.

File: node/pkg/adminrpc/adminserver.go (hunk 1 above)
Comment (johnsaigle) [reply node/pkg/adminrpc/adminserver.go:350]: I get that the implementation of these parsing/decoding functions is to allow setting addresses of variable length, but somewhere along the line we should assert that the addresses have a length of either:
* 0 (unset)
* 20 (EVM-like)
* 32 (most other chains)
* ... any other fixed types we're aware of

Otherwise we open the opportunity to fat-finger one of these addresses by 1 character and only discover that on-chain or when the Guardians are unable to produce the same governance VAA.

It probably makes sense to add this check to `Serialize()` since everything will go through there eventually.

File: node/pkg/adminrpc/prototext_test.go [hunk 3]
```diff
@@ -625,6 +625,160 @@ var adminCommandTest = []adminCommandTestEntry{
...
+				messages: {
+					sequence: 1
+					nonce: 2
+					bridge_set_pauser_addresses: {
+						module: "TokenBridge"
+						target_chain_id: 2
+						unpauser: "Hello, World!"
+					}
+				}`,
+	},
```
Comment (johnsaigle) [inline node/pkg/adminrpc/prototext_test.go:780]: We should add failing checks here for addresses of odd-length or other unsupported values like 18/22 and 31/33 (close to EVM or SVM length, but not quite).
Also we could add a test at the boundary of length 255 (happy path, max length) and 256 (uint8 overflow) if you want to keep support for very long address inputs.

File: sdk/vaa/payloads.go [hunk 4]
```diff
@@ -412,6 +430,21 @@ func (r BodyTokenBridgeUpgradeContract) Serialize() ([]byte, error) {
 	return serializeBridgeGovernanceVaa(r.Module, ActionUpgradeTokenBridge, r.TargetChainID, r.NewContract[:])
 }
 
+func (r BodyTokenBridgeSetPauserAddresses) Serialize() ([]byte, error) {
+	if len(r.Pauser) > math.MaxUint8 {
+		return nil, fmt.Errorf("pauser too long: %d bytes (max %d)", len(r.Pauser), math.MaxUint8)
```
Comment (johnsaigle) [inline sdk/vaa/payloads.go:438]: The max should probably be 32 here unless we're aware of chains that need support for longer addresses.

File: sdk/vaa/payloads.go [hunk 5]
```diff
@@ -412,6 +430,21 @@ func (r BodyTokenBridgeUpgradeContract) Serialize() ([]byte, error) {
 	return serializeBridgeGovernanceVaa(r.Module, ActionUpgradeTokenBridge, r.TargetChainID, r.NewContract[:])
 }
 
+func (r BodyTokenBridgeSetPauserAddresses) Serialize() ([]byte, error) {
```
Comment (johnsaigle) [inline sdk/vaa/payloads.go:433]: We could assert `r.Module == TokenBridge` here

File: node/pkg/adminrpc/adminserver.go (hunk 1 above)
Comment (johnsaigle) [reply node/pkg/adminrpc/adminserver.go:350]: Alternatively you could delegate all of this into Serialize() and skip the validation in the adminserver since Serialize() is checked and called here anyway using similar error messages.

Comment (evan-gray) [reply node/pkg/adminrpc/adminserver.go:350]: not sure I'm convinced this should change, at face value, mostly because I'm not sure what future chains may necessarily have fixed size addresses, such as Near and Cardano

File: node/pkg/adminrpc/prototext_test.go (hunk 3 above)
Comment (evan-gray) [reply node/pkg/adminrpc/prototext_test.go:780]: won't `hex.DecodeString` already reject odd-length hex?

File: sdk/vaa/payloads.go (hunk 4 above)
Comment (evan-gray) [reply sdk/vaa/payloads.go:438]: near uses longer keys (2-64 bytes). not sure if there will be others in the future, but I don't see a reason to arbitrarily limit this when we know there may be others we don't know about.

File: node/cmd/guardiand/admintemplate.go (hunk 2 above)
Comment (evan-gray) [reply node/cmd/guardiand/admintemplate.go]: updated the comment

File: sdk/vaa/payloads.go (hunk 5 above)
Comment (evan-gray) [reply sdk/vaa/payloads.go:433]: updated.

File: node/pkg/adminrpc/prototext_test.go (hunk 3 above)
Comment (johnsaigle) [reply node/pkg/adminrpc/prototext_test.go:780]: Yes, oops. With your other comment about not enforcing fixed-lengths, this is resolved.

**Review by johnsaigle — APPROVED** [review]

**Review by panoel — APPROVED** [review]
